### ISSUE TITLE
Fix conversions between NSArray and vector

### ISSUE THREAD

**Issue by nguyenhuy** [facebook/AsyncDisplayKit#556]
Follow-up issue after #468:

There are some places (e.g: ASStackLayoutSpec) where an array is created and passed to a method, just to be unpacked into a C++ structure (i.e: std::vector). This trick is used to bridge between the (pure Objective-C) public API and the (C++ heavy) internal implementation.

As @appleguy suggested, perhaps we should provide an internal API for those areas to use a fast-path.

**Comment by appleguy** [comment]
@nguyenhuy - The community is planning an exciting long term road map for the project and getting organized around how to deliver these feature requests. We are moving to a new issue tracker with more sophisticated planning capabilities (check out www.realArtists.com and their new product: Ship!). 

If you are interested in helping contribute to this component or any other, don’t hesitate to send me an email at AsyncDisplayKit@gmail.com. If you would like to contribute for a few weeks, I can also add you to our Ship bug tracker so that you can see what everyone is working on and actively coordinate with us. 

As always, keep filing issues and submitting pull requests here on Github and we will only move things to the new tracker if they require long term coordination.
